server/src/main/java/de/ingrid/igeserver/features/ogc_api_records/services/research_query/OgcApiResearchQuery.kt

```diff
@@ -66,36 +68,26 @@ abstract class OgcApiResearchQuery {
         clausesList.add(BoolFilter("OR", listOf("exceptFolders"), null, null, true))
 
         clausesList.add(BoolFilter("OR", listOf("document1.state = 'PUBLISHED'"), null, null, false))
-        // bbox // check if 4 values is true
-        if (ogcParameter.bbox != null) {
-            val boundingBox = ogcParameter.bbox.map { coordinate -> coordinate.toString() }
-            clausesList.add(BoolFilter("OR", listOf("ingridSelectSpatial"), null, boundingBox, true))
-        }
-        // time span
-        if (ogcParameter.datetime != null) {
-            val dateList = ogcDateTimeConverter(ogcParameter.datetime)
+
+        ogcParameter.datetime?.let { datetime ->
+            val dateList = ogcDateTimeConverter(datetime)
             clausesList.add(BoolFilter("OR", listOf("selectTimespan"), null, dateList, true))
         }
-        // filter by doc type
-        if (ogcParameter.type != null) {
+
+        ogcParameter.type?.let { type ->
             val typeList = mutableListOf<String>()
-            for (name in ogcParameter.type) {
-                typeList.add("document_wrapper.type = '$name'")
-            }
+            for (name in type) typeList.add("document_wrapper.type = '$name'")
             clausesList.add(BoolFilter("OR", typeList, null, null, false))
         }
 
-        val profileSpecificClausesList = profileSpecificClauses()
-
-        if (profileSpecificClausesList != null) {
-            clausesList.addAll(profileSpecificClausesList)
-        }
+        profileSpecificClauses()?.let { clausesList.addAll(it) }
 
         return clausesList
     }
 
     fun createQuery(ogcFilterParameter: OgcFilterParameter): ResearchQuery {
         ogcParameter = ogcFilterParameter
```

Comment:
Ich sehe hier noch ein Problem!
Die ResearchQuery-Klassen sind alles Services und werden nicht neu instantiiert. Allerdings wird in ihnen ein Zustand gesetzt (ogcParameter). Wenn nun ein paralleler Request kommt mit anderen Parametern, dann kommt es zu Problemen, da `ogcParameter` gemeinsam genutzt wird. In solch einem Fall muss die Factory eine neue Instanz zurückgeben. 

Jetzt kann die Factory angepasst werden, dass explizit zu einem Profil eine bestimmte Query-Klasse verwendet wird (dann werden die Klassen nicht mehr als Beans annotiert) oder du behältst die aktuelle Logik bei, aber erzeugst eine Wrapper-Klasse, die die Parameter und die Query-Klasse enthält. Die Query-Klasse muss dann über den Wrapper aufgerufen werden, um die definierten Parameter mitzuteilen.